src/System.Drawing.Common/src/System/Drawing/Icon.Windows.cs

```diff
@@ -159,42 +160,36 @@ void ISerializable.GetObjectData(SerializationInfo si, StreamingContext context)
 
         public static Icon ExtractAssociatedIcon(string filePath) => ExtractAssociatedIcon(filePath, 0);
 
-        private static Icon ExtractAssociatedIcon(string filePath, int index)
+        private unsafe static Icon ExtractAssociatedIcon(string filePath, int index)
         {
             if (filePath == null)
             {
                 throw new ArgumentNullException(nameof(filePath));
             }
 
-            Uri uri;
-            try
-            {
-                uri = new Uri(filePath);
-            }
-            catch (UriFormatException)
-            {
-                // It's a relative pathname, get its full path as a file. 
-                filePath = Path.GetFullPath(filePath);
-                uri = new Uri(filePath);
-            }
-
-            if (!uri.IsFile)
-            {
-                return null;
-            }
-
+            filePath = Path.GetFullPath(filePath);
             if (!File.Exists(filePath))
             {
-                throw new FileNotFoundException(filePath);
+                throw new FileNotFoundException(new FileNotFoundException().Message, filePath);
```

Comment:
Why is using the default's constructor message better than the one that gets constructed for when the message is null?